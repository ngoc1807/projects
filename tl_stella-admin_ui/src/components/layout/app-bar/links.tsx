import { FC, useState } from "react";
import { Menu, MenuItem, Button } from "@mui/material";
import { KeyboardArrowDown as KeyboardArrowDownIcon } from "@mui/icons-material";
import { ComingSoonTooltip } from "../sub-app-bar/coming-soon-tooltip";

export const Links: FC = () => {
  const [accountMenuElement, setAccountMenuElement] = useState<
    undefined | HTMLElement
  >();
  const openAccountMenu = Boolean(accountMenuElement);
  // const handleAccountMenuClick = (
  //   event: MouseEvent<HTMLButtonElement>
  // ) => {
  //   setAccountMenuElement(event.currentTarget);
  // };
  const handleAccountMenuClose = () => {
    setAccountMenuElement(undefined);
  };

  const [billingMenuElement, setBillingMenuElement] = useState<
    undefined | HTMLElement
  >();
  const openBillingMenu = Boolean(billingMenuElement);
  // const handleBillingMenuClick = (
  //   event: MouseEvent<HTMLButtonElement>
  // ) => {
  //   setBillingMenuElement(event.currentTarget);
  // };
  const handleBillingMenuClose = () => {
    setBillingMenuElement(undefined);
  };
  // aboveSmallBreakpoint &&
  return (
    <>
      <ComingSoonTooltip>
        <Button
          color="inherit"
          // onClick={handleAccountMenuClick}
          sx={{ cursor: "default" }}
          disableRipple
        >
          Account <KeyboardArrowDownIcon fontSize="small" />
        </Button>
      </ComingSoonTooltip>
      <Menu
        anchorEl={accountMenuElement}
        open={openAccountMenu}
        onClose={handleAccountMenuClose}
        transformOrigin={{ horizontal: "right", vertical: "top" }}
        anchorOrigin={{ horizontal: "right", vertical: "bottom" }}
      >
        <MenuItem onClick={handleAccountMenuClose}>Option 1</MenuItem>
        <MenuItem onClick={handleAccountMenuClose}>Option 2</MenuItem>
        <MenuItem onClick={handleAccountMenuClose}>Option 3</MenuItem>
      </Menu>
      <ComingSoonTooltip>
        <Button
          color="inherit"
          // onClick={handleBillingMenuClick}
          sx={{ cursor: "default" }}
          disableRipple
        >
          Billing <KeyboardArrowDownIcon fontSize="small" />
        </Button>
      </ComingSoonTooltip>
      <Menu
        anchorEl={billingMenuElement}
        open={openBillingMenu}
        onClose={handleBillingMenuClose}
        transformOrigin={{ horizontal: "right", vertical: "top" }}
        anchorOrigin={{ horizontal: "right", vertical: "bottom" }}
      >
        <MenuItem onClick={handleBillingMenuClose}>Option 1</MenuItem>
        <MenuItem onClick={handleBillingMenuClose}>Option 2</MenuItem>
        <MenuItem onClick={handleBillingMenuClose}>Option 3</MenuItem>
      </Menu>
    </>
  );
  // : null;
};
