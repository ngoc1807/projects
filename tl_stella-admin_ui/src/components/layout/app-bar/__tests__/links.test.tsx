import { cleanup, render } from "@testing-library/react";
import { Links } from "../links";

describe("links", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly", () => {
    const { container } = render(<Links />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
