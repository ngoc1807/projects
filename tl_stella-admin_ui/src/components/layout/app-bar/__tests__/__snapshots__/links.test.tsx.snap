// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`links should render correctly 1`] = `
<button
  aria-label="Coming Soon!"
  class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-wpg80s-MuiButtonBase-root-MuiButton-root"
  data-mui-internal-clone-element="true"
  tabindex="0"
  type="button"
>
  Account 
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
    data-testid="KeyboardArrowDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
    />
  </svg>
</button>
`;
